packages/Main/src/Renderer/WebGLComposer.ts

```diff
@@ -0,0 +1,223 @@
+import * as THREE from 'three';
+import { RasterTile } from './RasterTile';
+
+// shader for copying a 2D texture to a framebuffer
+const copyTextureShader = {
+    vertexShader: `
+        varying vec2 vUv;
+        void main() {
+            vUv = uv;
+            gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.0);
+        }
+    `,
+    fragmentShader: `
+        precision highp float;
+        uniform sampler2D sourceTexture;
+        varying vec2 vUv;
+        void main() {
+            gl_FragColor = texture2D(sourceTexture, vUv);
+        }
+    `,
+};
+
+let renderTarget: THREE.WebGLRenderTarget | null = null;
+let material: THREE.ShaderMaterial | null = null;
+let quadScene: THREE.Scene | null = null;
+let quadCam: THREE.OrthographicCamera | null = null;
```

Comment:
Since this is constant across the different calls to `drawTextureLayers` and is just a bunch of properties, you could make it local to the module.
```suggestion
const quadCam: THREE.OrthographicCamera = new THREE.OrthographicCamera(-1, 1, 1, -1, 0, 1);;
```